src/InlineText.php

```diff
@@ -7,4 +7,34 @@
  */
 class InlineText extends Text
 {
+
+    public function __construct($text = "", $markdownTag = "")
+    {
+        parent::__construct($text);
+        $this->mdTag = $markdownTag;
+    }
+
+    public function write()
+    {
+        $tag = parent::write($this->text);
+        if (!trim($tag)) return $tag;
+        $tag = trim($tag);
```

Comment:
Why do we trim whitespace here and not below where the trimmed value is used?

This has the effect that `\n\n\n A \n\n\n` becomes `A` but `\n\n\n             \n\n\n` is returned as-is. It seems arbitrary from the user's perspective that changing a single character has such an effect.

Reply:
Oh, I thought this was the desired outcome?

Reply:
@graycodes I don't think this was how it worked previously. This way messes up the fairly common case of having leading/trailing whitespace in XML tags:
```xml
<strong>my strong
text </strong>is strong
```
becomes:
```md
**my strong**
**text**is strong
```
instead of:
```md
**my strong**
**text** is strong
```

Reply:
Oooooh k

Reply:
@graycodes I'm wondering if it would be easier to just re-write the `outerspace` API in PHP and do a more direct translation of the Tugboat code to ensure we don't have regressions like this being introduced. We've been through the process of fixing the many corner cases and simplifying logic already with the CopilotTags rewrite so I don't see the value in changing the logic again now.

Reply:
From what I've seen it's mostly fine, there are just a few things that have changed that are concerning me!

Reply:
Wasn't an intentional change of logic. I thought I had all the tests covered but clearly there's some gaps

Reply:
@graycodes yeah, gaps in tests was definitely a big issue that I'm hoping we can remedy in the PHP version.. I realise it makes it harder to validate that the PHP version works the same way as before. 😅